torch/csrc/jit/passes/onnx/function_extraction.cpp

```diff
@@ -1020,6 +1071,28 @@ NodeAttrNameMap FunctionExtractor::run() {
   return node_attr_to_name;
 }
 
+Node* GetNodeOfMostRecentScope(Node* forward_node) {
+  // This function is used to retrieve the node representing the most recent
+  // ScopePtr. This function should only be invoked from module forward hook. At
+  // this point, module forward call is completed, and the most recent ScopePtr
+  // is popped from TracingState.
+  // This function inspects the node, and its subblock, to find
+  // the node associated with the most recent ScopePtr.
+
+  // Skip the "real" last node which is `return_node`.
+  TORCH_INTERNAL_ASSERT(forward_node->kind() == prim::TracedModuleForward);
```

Comment:
Can we log something about `forward_node` here so the error message is useful in case this fails?